harness/libraries/influx_handler.py

```diff
@@ -0,0 +1,315 @@
+
+
+class influx_handler:
+    """
+        Abstracts the handling of InfluxDB requests
+    """
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+    #                                                                 @
+    # Special methods                                                 @
+    #                                                                 @
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+    def __init__(self, logger=None):
+        # Inherit logger from the class that initializes this class
+        if not logger:
+            print("logger parameter not provided to influx_handler. Defaulting to stdout")
+            self.logger = default_logger()
+        else:
+            self.logger = logger
+
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+    #                                                                 @
+    # End of special methods                                          @
+    #                                                                 @
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+    #                                                                 @
+    # Public methods.                                                 @
+    #                                                                 @
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+    def post(self, msg, url=None, token=None):
+        """
+            Posts an event to InfluxDB
+        """
+        # Failed import of requests is non-fatal, since InfluxDB is an extension. Check to make sure it imported
+        if not 'requests' in sys.modules:
+            self.logger.doWarningLogging(f"'requests' is not in sys.modules. InfluxDB logging is disabled. Events can be logged after the run using --mode influx_log.")
+            self.logger.doInfoLogging(f"Skipping message: {msg}.")
+            return False
+        # Check if InfluxDB is explicitly disabled:
+        if 'RGT_DISABLE_INFLUX' in os.environ and str(os.environ['RGT_DISABLE_INFLUX']) == '1':
+            self.logger.doInfoLogging("InfluxDB is explicitly disabled with RGT_DISABLE_INFLUX=1")
+            return False
+        # Required variables:
+        if not (('RGT_INFLUX_URI' in os.environ or url) and ('RGT_INFLUX_TOKEN' in os.environ or token)):
+            # Then we don't know where/how to post
+            self.logger.doWarningLogging('RGT_INFLUX_URI and RGT_INFLUX_TOKEN are required environment variables to enable InfluxDB logging, \
+                                            or the url and token parameters must be specified')
+            return False
+        influx_url = url if url else os.environ['RGT_INFLUX_URI']
+        # Checks to make sure the URL is properly formatted
+        influx_url = self._check_url_endpoint(influx_url, read=False)
```

Comment:
This line makes me hesitate a little. Basically, regardless of where it receives the influx_url from (envvar or named arg), it checks the endpoint, and formats it as an InfluxDB API v2 write request. InfluxDB versions <2.0 will not be able to understand this. It's possible that we say InfluxDB v2+ is required in order to use this feature.

The reason I have this check_url_endpoint method is that the influx URL could potentially be provided as just a standard hostname (ie https://influx.abc.xyz) instead of https://influx.abc.xyz/api/v2/write?db=accept, or it could be provided with the /api/v2/write endpoint, when /api/v2/query is needed (in the event of query_flux).